src/api/routes/auth.ts

```diff
@@ -39,7 +47,13 @@ export default () => {
 
 function handleError(error: Error) {
   if (error instanceof AuthorisationFailedError) {
-    throw new ApiError(403, error.failureReason.toString());
+    throw new ApiError(422, error.failureReason.toString());
+  }
+  if (error instanceof ResourceNotFoundError) {
+    throw new ApiError(422, `${error.resourceName}.not-found`);
+  }
+  if (error instanceof DomainValidationError) {
+    throw new ApiError(422, `invalid.${error.field}`, error.reason);
```

Comment:
We repeat pretty much the same code over multiple endpoints, should we centralise?

Reply:
I thought of it. For now I'm erring towards having duplication and being sure that the status codes and error codes fit the endpoint being called. As you said it's not exactly the same but "pretty much" the same. Some of the response codes are different based on the actual endpoints under which the same exception is triggered.  